core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -833,12 +832,17 @@ class SnappyParser(session: SnappySession)
               (args(0) == -10) || (args(0) == -1 && (index & 0x1) == 1) ||
               // all even args
               (args(0) == -2 && (index & 0x1) == 0) =>
+            l match {
+              case pl: ParamLiteral  if pl.tokenized => throw new ParseException(
+                s"function $fnName cannot have parameterized arguments")
```

Comment:
I think this should be "if pl.tokenized && isPreparePhase". Also the error message can include the specific parameter number: "Function $fnName cannot have parameterized argument ${index + 1}"

Reply:
ok